crates/blockifier/src/execution/contract_class.rs

```diff
@@ -571,26 +588,23 @@ impl SierraContractClassV1 {
     }
 }
 
-impl TryFrom<SierraContractClass> for SierraContractClassV1 {
-    // TODO error type
-    type Error = ProgramError;
-
-    fn try_from(class: SierraContractClass) -> Result<Self, Self::Error> {
-        // todo(rodro): we are having two instances of a sierra program, one it's object form
-        // and another in its felt encoded form. This can be avoided by either:
-        //   1. Having access to the encoding/decoding functions
-        //   2. Refactoring the code on the Cairo mono-repo
-        Ok(Self(Arc::new(SierraContractClassV1Inner {
-            sierra_program: class.extract_sierra_program().unwrap(),
-            entry_points_by_type: class.entry_points_by_type,
-            sierra_program_raw: class.sierra_program,
-        })))
-    }
-}
-
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Debug)]
 pub struct SierraContractClassV1Inner {
+    // todo(xrvdg) can we make sierra_program private such that it
+    // is only available for debugging purposes?
+    // For now execute_entry_point_call needs it
+    // to get the entry function
```

Comment:
In line with a previous comment, when we have a the compilation outside the Blockifier, would it be possible to calculate the entry point there?

Reply:
`ContractClassV0Inner` and `ContractClassV1Inner` do store the entry points so it makes sense to do it for native contract class as well.  Think that best done in a follow up PR.